xslt/Channels.xsl

```diff
@@ -37,49 +45,94 @@
         </xsl:if>
     </xsl:template>
 
-
+    <xsl:template match="DetectorSettings">
+        <xsl:element name="ome:DetectorSettings">
+            <xsl:attribute name="ID">
+                <xsl:choose>
+                    <xsl:when test="substring(Detector/@Id, 0, 17)='Detector:Camera '">
+                        <xsl:text>Detector:</xsl:text>
+                        <xsl:value-of select="substring(Detector/@Id, 17, 1)"/>
+                    </xsl:when>
+                    <xsl:otherwise>
+                        <xsl:value-of select="Detector/@Id"/>
+                    </xsl:otherwise>
+                </xsl:choose>
+            </xsl:attribute>
+            <xsl:attribute name="Binning">
+                <xsl:choose>
+                    <xsl:when test="Binning='other'">
+                        <xsl:value-of select="Binning"/>
+                    </xsl:when>
+                </xsl:choose>
+                <xsl:value-of select="substring(Binning, 0, 2)"/>
+                <xsl:text>x</xsl:text>
+                <xsl:value-of select="substring(Binning, 0, 2)"/>
+            </xsl:attribute>
+        </xsl:element>
+    </xsl:template>
 
     <!-- /Metadata/DisplaySetting/Channels/Channel => /OME/Image/Channel -->
     <!-- /Metadata/DisplaySetting/Channels/Channel/DyeName => /OME/Image/Channel@Name -->
     <!-- /ImageDocument/Metadata/Information/Image/Dimensions/Channels/Channel/ExcitationWavelength => /Ome/Image/Channel@ExcitationWavelength  -->
     <!-- /ImageDocument/Metadata/Information/Image/Dimensions/Channels/Channel/EmissionWavelength => /Ome/Image/Channel@EmissionWavelength  -->
     <xsl:template match="Channel">
         <xsl:param name="info_channel"/>
-        <xsl:element name="Channel">
+        <xsl:param name="idx"/>
+        <xsl:element name="ome:Channel">
+            <xsl:attribute name="ID">
+                <xsl:value-of select="$info_channel/@Id"/>
+                <xsl:text>-</xsl:text>
+                <xsl:value-of select="$idx"/>
+            </xsl:attribute>
             <xsl:attribute name="Name">
                 <xsl:value-of select="@Name"/>
             </xsl:attribute>
             <xsl:attribute name="AcquisitionMode">
                 <xsl:value-of select="$info_channel/AcquisitionMode"/>
             </xsl:attribute>
             <xsl:apply-templates select="IlluminationType"/>
-                <xsl:apply-templates select="$info_channel/ExcitationWavelength"/>
-                <xsl:apply-templates select="$info_channel/EmissionWavelength"/>
-            <xsl:attribute name="Fluorophore">
-                <xsl:value-of select="DyeName"/>
-            </xsl:attribute>
+            <xsl:apply-templates select="$info_channel/ExcitationWavelength"/>
+            <xsl:apply-templates select="$info_channel/EmissionWavelength"/>
+<!--            <xsl:attribute name="Fluorophore">-->
+<!--                <xsl:value-of select="DyeName"/>-->
+<!--            </xsl:attribute>-->
```

Comment:
This might map to the @Fluor attribute in channel I'm not sure thought. I can't find Fluorophore in the OME spec.

Reply:
Yeah, this does seem like it maps to `@Fluor`, and that validates properly.

Reply:
Resolved in commit 9ee7971c2dea4c4b64ca97158befe3dcf84fb243.